tests/Unit/HtmlProcessor/AbstractHtmlProcessorTest.php

```diff
@@ -732,4 +746,42 @@ public function getDomDocumentVoidElementNotHasChildNodes(string $htmlWithNonXml
             self::assertFalse($element->hasChildNodes());
         }
     }
+
+    /**
+     * Asserts that two HTML strings are equal, allowing for whitespace differences in the HTML element itself (but not
+     * its descendants) and after its closing tag.
+     *
+     * @param string $expected
+     * @param string $actual
+     * @param string $message
+     */
+    private static function assertEqualsHtml(string $expected, string $actual, string $message = '')
+    {
+        $normalizedExpected = self::normalizeHtml($expected);
+        $normalizedActual = self::normalizeHtml($actual);
+
+        self::assertSame($normalizedExpected, $normalizedActual, $message);
+    }
+
+    /**
+     * Normalizes whitespace in the HTML element itself (but not its descendants) and after its closing tag, with a
+     * single newline inserted or replacing whitespace at positions where whitespace may occur but is superfluous.
+     *
+     * @param string $html
+     *
+     * @return string
+     */
+    private static function normalizeHtml(string $html)
```

Comment:
I propose renaming this to `normalizeHtmlElement` to make it clear it's not about the string `$html`, but about the specific element. WDYT?

Reply:
OK.  Have changed thus.  The original name was because I wasn't sure whether some whitespace within the `<head>` element might also need normalizing, but testing against changes in #831 I found this wasn't required.

Reply:
I think the callers still are using the old name.

Reply:
> I think the callers still are using the old name.

Sorry, I'd fixed this in a local copy where I ran the tests, but forgot to merge with the `git` folder from which I did the `push`.